Reusing scripts for live resources & no more deploy buttons (#8398)
Azure/azure-sdk-for-js: sdk/eventhub/event-hubs/test/README.md

# Testing

To test this project, make sure to first build it properly by following our [building instructions](https://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md#building). Once the project is correctly built, you will be able to run the tests by following the [testing instructions](https://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md#testing).

> The environment variables you will need to properly assign to run the live test of this project are the following:

Reviewer: ```suggestion
The Event Hubs project does not have any recorded tests and so, all the tests require an Azure Event Hubs namespace to be set up beforehand with at lease a single Event Hub instance in it. Set the below environment variables to run the tests
```

Reviewer: Use similar format for 
- event-processor-host
- checkpointstore
- service bus

Author: on it

> - `EVENTHUB_NAME`: The name of your Azure Event Hub account.

Reviewer: ```suggestion
- `EVENTHUB_NAME`: The name of your Azure Event Hub instance, not the Event Hub namespace.
```

> - `EVENTHUB_NAME`: The name of your Azure Event Hub account.
> - `EVENTHUB_CONNECTION_STRING`: The connection string of your Azure Event Hub account.

Reviewer: ```suggestion
- `EVENTHUB_NAME`: The name of your Azure Event Hub instance, not your Azure Event Hubs namespace.
- `EVENTHUB_CONNECTION_STRING`: The connection string of your Azure Event Hubs namespace.
```

> - `IOTHUB_CONNECTION_STRING`: The connection string of your Azure IoT Hub account.

Reviewer: This isnt used by Track 2 event hubs, so we dont need it specified here
```suggestion
```

> - `IOTHUB_EH_COMPATIBLE_CONNECTION_STRING`: An Azure Event Hub compatible connection string provided by your Azure IoT Hub account.

Reviewer: @chradek This is being used in the `iothub.spec.ts` file which I now believe we should get rid of entirely. At the end of the day, we dont treat an EH compatible IotHub connection string any different than the normal EH connection string, so what are we even testing here? In Track 1, this made sense because we were testing with the input being the Iothub connection string. Thoughts?

Author: I've removed this preemptively

Reviewer 2: Yes, I think this is fair to remove.

> - `STORAGE_CONNECTION_STRING`: The connection string of an Azure Storage account.

Reviewer: This is used in the checkpoint store project, not event hubs, so we can remove this too.

```suggestion
```

> Keep in mind that this project **does not use [the recorder](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/test-utils/recorder)** yet.

Reviewer: We should mention this upfront. https://github.com/Azure/azure-sdk-for-js/pull/8398/files?diff=split&w=1#r411887199 should take care of it.

```suggestion
```

The Azure resources that are used by the tests in this project are:

- An [Azure Event Hubs namespace](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#namespace).
- Some very basic Network configurations for the Event Hubs namespace.
- A [consumer group](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#consumer-groups) to publish and subscribe to the Event Hubs namespace created.
- An [Azure Storage account](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview) configured to provide [blob storage](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#blob-storage-resources).
- An [Azure IoT Hub account](https://docs.microsoft.com/en-us/azure/iot-hub/about-iot-hub).

You will also need to set the below environment variables to run the live tests:

- `TEST_MODE`: Should have `live` assigned.

> - `EVENTHUB_NAME`: The name of your Azure Event Hub account.

Reviewer 3: I don't think it's the name of the "account".

# Testing

To run our tests, first do a full update of the project by running `rush update`, then rebuild either this package, with `rush build -t @azure/event-hubs` or all packages with `rush build`, then make sure to build the tests with `rush build:test`. After that, you can either run our unit tests, which won't reach to live services, and our integration tests, which will indeed require existing, reachable resources.

## Unit tests

To run unit tests, go to the root of this library and run `rushx unit-test`.

> This will run tests in a purely nodejs environment, and in a browser environment using [Karma](https://karma-runner.github.io/latest/index.html).

Author: ```suggestion
This will run tests in NodeJS and in a browser (through [Karma](https://karma-runner.github.io/latest/index.html)).
```

You won't need to have any environment configuration, nor internet access, for these tests to run.

Keep in mind that if you modify the test files, the API shape of this library, or the underlying network calls, the recordings need to be re-generated.
You can read more in the section: [Regenerating recordings](#regenerating-recordings).

## Run all tests

Run `rushx test:node` or `rushx test:browser` from your terminal to run all the tests under the `./test` folder

## Run all tests in a single test suite

Append the `.only` on the `describe` method corresponding to the test suite.

Then run `rushx test:node` or `rushx test:browser` from your terminal.

## Run a single test

Append the `.only` on the `it` method corresponding to the test.

Then run `rushx test:node` or `rushx test:browser` from your terminal.

## Debug tests using Visual Studio Code

- Select the debug configuration `Debug Unit Tests` from the drop down in the debug viewlet.
- Add breakpoints in the test you want to debug. Tip: It is helpful to append the `.only` on the test method if you want only that method to be run.
- Start Debugging

## Unit tests

> To run unit tests, go to the root of this library and run `rushx unit-test`. This will run tests in a purely nodejs environment, and in a browser environment using [Karma](https://karma-runner.github.io/latest/index.html). These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses. You won't need to have any environment configuration, nor internet access, for these tests to run.

Author: ```suggestion
To run unit tests, go to the root of this library and run `rushx unit-test`. This will run tests in NodeJS and in a browser (through [Karma](https://karma-runner.github.io/latest/index.html)). These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses. You won't need to have any environment configuration, nor internet access, for these tests to run.
```
